File: frontend/src/types/game.ts
export interface ExtremeGameResponse {
  round_id: number;
  ability_description: string;
  current_score: number;
  expires_at: string;
  server_now: string;
}

export type AbilityKey = "passive" | "q" | "w" | "e" | "r";

export interface ExtremeGameUserAnswer {
  round_id: number;
  champion_name: string;
  ability_id: AbilityKey | null;
}

export interface ExtremeGameCheckRound {
  correct: boolean;
  correct_champion_name: string;
  correct_ability_id: string;
  correct_champion_spell_icon: string;
  correct_champion_id: string;
  correct_champion_image: string;
  current_score: number;
  timeout: boolean;
  next_round: ExtremeGameResponse | null;
}

export interface ChampionSearch {
  id: string;
  champ_name: string;
  champ_icon: string;
}

File: backend/app/schemas/game.py
from pydantic import BaseModel
from app.schemas.common import UTCDateTime
from app.models.game import AbilityKey


class ExtremeGameCreateRound(BaseModel):
    round_id: int
    ability_description: str
    current_score: int
    expires_at: UTCDateTime
    server_now: UTCDateTime


class ExtremeGameUserAnswer(BaseModel):
    round_id: int
    champion_name: str
    ability_id: AbilityKey


class ExtremeGameCheckRound(BaseModel):
    correct: bool
    correct_champion_name: str
    correct_ability_id: AbilityKey
    correct_champion_spell_icon: str
    correct_champion_image: str
    correct_champion_id: str
    current_score: int
    timeout: bool
    next_round: ExtremeGameCreateRound | None = None


class ExtremeGameNotFound(Exception):
    pass


class ExtremeGameRoundNotFound(Exception):
    pass
